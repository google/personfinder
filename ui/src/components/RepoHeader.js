import React, {Component} from 'react';
import {FormattedHTMLMessage, defineMessages, injectIntl} from 'react-intl';
import {Link} from "react-router-dom";

const messages = defineMessages({
  productName: {
    id: 'RepoHeader.productName',
    defaultMessage: 'Google Person Finder',
    description: ('Name of the product. Person Finder is a tool that helps '
        + 'people reconnect with friends/family after a disaster.'),
  },
});

class RepoHeader extends Component {
  render() {
    return (
      <div id="repoheader">
        <div id="repoheader-backbutton">
          <Link to={"/" + this.props.repoInfo.repoId}>
            <div><img src="/static/icons/maticon_arrow_back.svg" /></div>
          </Link>
        </div>
        <div id="repoheader-info">
          <p className="mdc-typography--subtitle1">
            <FormattedHTMLMessage {...messages.productName} />
          </p>
          <p className="mdc-typography--subtitle2">
            {this.props.repoInfo.title}
          </p>
        </div>
      </div>
    );
  }
}

export default injectIntl(RepoHeader);
