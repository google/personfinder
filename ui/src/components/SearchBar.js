/*
 * Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, {Component} from 'react';
import {defineMessages, injectIntl} from 'react-intl';
import {withRouter} from 'react-router-dom';
import TextField, {HelperText, Input} from '@material/react-text-field';

const MESSAGES = defineMessages({
  searchForAPerson: {
    id: 'SearchBar.searchForAPerson',
    defaultMessage: 'Search for a person',
    description: 'Label on a form for searching for information about someone.',
  },
});

class SearchBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: (props.initialValue == null ? '' : props.initialValue)};
    this.handleKeyDown = this.handleKeyDown.bind(this);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.initialValue != this.props.initialValue) {
      this.setState({value: this.props.initialValue});
    }
  }

  handleKeyDown(e) {
    if (e.key === 'Enter') {
      e.preventDefault();
      this.props.onSearch(this.state.value);
    }
  }

  render() {
    return (
        <div className='searchbar-wrapper'>
          <TextField
            label={this.props.intl.formatMessage(MESSAGES.searchForAPerson)}
            outlined
            className='searchbar'
          >
            <Input
              value={this.state.value}
              onKeyDown={this.handleKeyDown}
              onChange={(e) => this.setState({value: e.target.value})} />
          </TextField>
        </div>
    );
  }
}

export default injectIntl(SearchBar);
