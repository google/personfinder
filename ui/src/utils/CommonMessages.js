/*
 * Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {defineMessages} from 'react-intl';

const COMMON_MESSAGES = defineMessages({
  no: {
    id: 'CommonMessages.no',
    defaultMessage: 'No',
    description: 'A negative answer to a yes/no question.',
  },
  yes: {
    id: 'CommonMessages.yes',
    defaultMessage: 'Yes',
    description: 'An affirmative answer to a yes/no question.',
  },
});

export default COMMON_MESSAGES;
