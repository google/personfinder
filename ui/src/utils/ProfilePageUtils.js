/*
 * Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {defineMessages} from 'react-intl';

const MESSAGES = defineMessages({
  facebook: {
    id: 'ProfilePageUtils.facebook',
    defaultMessage: 'Facebook',
    description: 'The social media site.',
  },
  twitter: {
    id: 'ProfilePageUtils.twitter',
    defaultMessage: 'Twitter',
    description: 'The social media site.',
  },
  linkedin: {
    id: 'ProfilePageUtils.linkedin',
    defaultMessage: 'LinkedIn',
    description: 'The networking site.',
  },
  otherWebsite: {
    id: 'ProfilePageUtils.otherWebsite',
    defaultMessage: 'Other website',
    description: ('A label for a button for users to add a link to a site '
        + 'other than a site on a pre-defined list.'),
  },
});

/**
 * A utility for people's profile pages.
 */
const ProfilePageUtils = Object.freeze({
  /**
   * A mapping from a profile page site ID to a MessageDescriptor of the name
   * of the site.
   */
  SITES: Object.freeze({
    'facebook': MESSAGES.facebook,
    'twitter': MESSAGES.twitter,
    'linkedin': MESSAGES.linkedin,
    'other': MESSAGES.otherWebsite,
  })
});

export default ProfilePageUtils;
