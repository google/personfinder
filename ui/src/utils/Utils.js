/*
 * Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import queryString from 'query-string';

/*
 * A class for utility functions.
 */
class Utils {

  static getURLParam(props, paramName) {
    return queryString.parse(props.location.search)[paramName];
  }

  static onExternalScriptLoad(url) {
    EXTERNAL_SCRIPTS[url].loaded = true;
    const callbacksList = EXTERNAL_SCRIPTS[url].callbacks;
    for (var i = 0; i < callbacksList.length; i++) {
      callbacksList[i]();
    }
  }

  static loadExternalScript(url, callback) {
    if (EXTERNAL_SCRIPTS[url] == undefined) {
      EXTERNAL_SCRIPTS[url] = {callbacks: [callback], loaded: false};
      const scriptTag = document.createElement('script');
      scriptTag.src = url;
      scriptTag.addEventListener(
          'load', () => Utils.onExternalScriptLoad(url));
      document.getElementsByTagName('body')[0].appendChild(scriptTag);
    } else if (EXTERNAL_SCRIPTS[url].loaded) {
      callback();
    } else {
      EXTERNAL_SCRIPTS[url].callbacks.push(callback);
    }
  }
}

export default Utils;
