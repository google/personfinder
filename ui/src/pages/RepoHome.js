/*
 * Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, {Component} from 'react';
import {FormattedMessage, defineMessages, injectIntl} from 'react-intl';
import {Link} from 'react-router-dom'
import Button from '@material/react-button';

import EndBarHeader from './../components/EndBarHeader.js';
import Footer from './../components/Footer.js';
import LoadingIndicator from './../components/LoadingIndicator.js';
import RepoHeader from './../components/RepoHeader.js';
import SearchBar from './../components/SearchBar.js';

const MESSAGES = defineMessages({
  howToUsePf: {
    id: 'RepoHome.howToUsePf',
    defaultMessage: 'How to use Person Finder',
    description: 'Anchor text for a link to a user guide.',
  },
  provideInfoAboutSomeone: {
    id: 'RepoHome.provideInfoAboutSomeone',
    defaultMessage: 'Provide information about someone',
    description: ('Label on a button for people who want to provide '
        + 'information about someone\'s status in the aftermath of a '
        + 'disaster.'),
  },
  or: {
    id: 'RepoHome.or',
    defaultMessage: 'Or',
    description: ('A heading for a section with one or more alternatives to '
        + 'the main option.'),
  },
  // TODO(nworden): confirm with UX that this should differ (begins with
  // "currently") from the similar message on the global homepage.
  repoRecordCount: {
    id: 'RepoHome.repoRecordCount',
    defaultMessage: 'Currently tracking {recordCount} records',
    description: ('A message displaying how many data records we have in the '
        + 'database.'),
  },
});

/*
 * A component for repo homepages.
 */
class RepoHome extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoaded: false,
      error: null,
      repo: null
    };
    this.repoId = this.props.match.params.repoId;
    this.goToCreate = this.goToCreate.bind(this);
    this.handleSearch = this.handleSearch.bind(this);
  }

  goToCreate() {
    this.props.history.push(`/${this.repoId}/create`);
  }

  handleSearch(query) {
    this.props.history.push({
        pathname: `/${this.repoId}/results`,
        search: '?query_name=' + encodeURIComponent(query),
      });
  }

  componentDidMount() {
    const apiUrl = `/${this.repoId}/d/repo`;
    fetch(apiUrl)
      .then(res => res.json())
      .then(
        (repo) => {
          this.setState({
            isLoaded: true,
            repo: repo
          });
        },
        (error) => {
          this.setState({
            isLoaded: true,
            error: error
          });
        }
      );
  }

  render() {
    if (this.state.error) {
      return <div>An error occurred</div>
    }
    if (!this.state.isLoaded) {
      return <LoadingIndicator />;
    }
    var recordCountContent = null;
    if (this.state.repo.recordCount > 0) {
      recordCountContent = (
          <p className='mdc-typography--body1 repohome-recordcount'>
            <FormattedMessage
              {...MESSAGES.repoRecordCount}
              values={{
                'recordCount': this.state.repo.recordCount
              }} />
          </p>
      );
    }
    return (
      <div>
        <RepoHeader
          repo={this.state.repo}
          backButtonTarget={'/'}
        />
        <div className='repohome-body'>
          <SearchBar
              repoId={this.repoId}
              initialValue=''
              onSearch={this.handleSearch} />
          {recordCountContent}
          <EndBarHeader>
            <FormattedMessage {...MESSAGES.or} />
          </EndBarHeader>
          <Button
            className='pf-button-secondary'
            outlined
            onClick={this.goToCreate}
          >
            {this.props.intl.formatMessage(MESSAGES.provideInfoAboutSomeone)}
          </Button>
        </div>
        <div className='org-boxwrapper repohome-footersection'>
          <p className='mdc-typography--body1 repohome-howtolink'>
            <Link to={{pathname: '/global/howto'}}>
              <FormattedMessage {...MESSAGES.howToUsePf} />
            </Link>
          </p>
          <Footer wrapped />
        </div>
      </div>);
  }
}

export default injectIntl(RepoHome);
