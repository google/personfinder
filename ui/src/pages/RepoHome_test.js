/*
 * Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */


import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import {createMemoryHistory} from 'history';
import React from 'react';
import {MemoryRouter} from 'react-router';

import RepoHome from './RepoHome';
import RepoHeader from '../components/RepoHeader';
import SearchBar from '../components/SearchBar';
import {mountWithIntl} from '../testing/enzyme-intl-helper';
import {flushPromises} from '../testing/utils';

Enzyme.configure({adapter: new Adapter()});

const REPO_DATA = {repoId: 'albany', title: 'Albany', recordCount: 100,};

function setupPageWrapper() {
  fetch.mockResponseOnce(JSON.stringify(REPO_DATA));
  const history = createMemoryHistory('/albany');
  const matchValue = {params: {repoId: 'albany'}};
  const wrapper = mountWithIntl(
    <MemoryRouter>
      <RepoHome history={history} match={matchValue} />
    </MemoryRouter>
  );
  return [wrapper, history];
}

describe('testing RepoHome', () => {
  beforeEach(() => {
    fetch.resetMocks();
  });

  test('RepoHeader configured correctly', () => {
    const [wrapper] = setupPageWrapper();
    return flushPromises().then(() => {
      wrapper.update();
      const actualRepoHeader = wrapper.find(RepoHeader).get(0);
      expect(actualRepoHeader.props.repo).toEqual(REPO_DATA);
      expect(actualRepoHeader.props.backButtonTarget).toBe('/');
      wrapper.unmount();
    });
  });

  test('SearchBar configured correctly', () => {
    const [wrapper, history] = setupPageWrapper();
    return flushPromises().then(() => {
      wrapper.update();
      const actualSearchBar = wrapper.find(SearchBar).get(0);
      expect(actualSearchBar.props.repoId).toBe('albany');
      expect(actualSearchBar.props.initialValue).toBe('');
      actualSearchBar.props.onSearch('thÃ¡tcher');
      expect(history.entries[1].pathname).toBe('/albany/results');
      expect(history.entries[1].search).toBe('?query_name=th%C3%A1tcher');
      wrapper.unmount();
    });
  });

  test('"Provide info" button goes to create page', () => {
    const [wrapper, history] = setupPageWrapper();
    return flushPromises().then(() => {
      wrapper.update();
      wrapper.find('.pf-button-secondary').at(0).simulate('click');
      expect(history.entries[1].pathname).toBe('/albany/create');
      wrapper.unmount();
    });
  });

  test('snapshot test for RepoHome', () => {
    // We don't use setupPageWrapper here because we need to avoid passing
    // a history object: it will generate random keys that mess up the
    // snapshot.
    fetch.mockResponseOnce(JSON.stringify(REPO_DATA));
    const matchValue = {params: {repoId: 'albany'}};
    const wrapper = mountWithIntl(
      <MemoryRouter>
        <RepoHome match={matchValue} />
      </MemoryRouter>
    );
    return flushPromises().then(() => {
      wrapper.update();
      expect(toJson(wrapper.find(RepoHome).at(0))).toMatchSnapshot();
      wrapper.unmount();
    });
  });
});
