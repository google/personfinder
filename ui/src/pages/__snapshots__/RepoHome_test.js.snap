// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing RepoHome snapshot test for RepoHome 1`] = `
<InjectIntl(RepoHome)
  match={
    Object {
      "params": Object {
        "repoId": "albany",
      },
    }
  }
>
  <RepoHome
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    match={
      Object {
        "params": Object {
          "repoId": "albany",
        },
      }
    }
  >
    <div>
      <InjectIntl(RepoHeader)
        backButtonTarget="/"
        repo={
          Object {
            "recordCount": 100,
            "repoId": "albany",
            "title": "Albany",
          }
        }
      >
        <RepoHeader
          backButtonTarget="/"
          intl={
            Object {
              "defaultFormats": Object {},
              "defaultLocale": "en",
              "formatDate": [Function],
              "formatHTMLMessage": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelative": [Function],
              "formatTime": [Function],
              "formats": Object {},
              "formatters": Object {
                "getDateTimeFormat": [Function],
                "getMessageFormat": [Function],
                "getNumberFormat": [Function],
                "getPluralFormat": [Function],
                "getRelativeFormat": [Function],
              },
              "locale": "en",
              "messages": Object {},
              "now": [Function],
              "onError": [Function],
              "textComponent": "span",
              "timeZone": null,
            }
          }
          repo={
            Object {
              "recordCount": 100,
              "repoId": "albany",
              "title": "Albany",
            }
          }
        >
          <div
            id="repoheader"
          >
            <div
              id="repoheader-backbutton"
            >
              <Link
                replace={false}
                to="/"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  <div>
                    <img
                      src="/static/icons/maticon_arrow_back.svg"
                    />
                  </div>
                </a>
              </Link>
            </div>
            <div
              id="repoheader-info"
            >
              <p
                className="mdc-typography--subtitle1"
              >
                <FormattedMessage
                  defaultMessage="Google Person Finder"
                  id="RepoHeader.productName"
                  values={Object {}}
                >
                  <span>
                    Google Person Finder
                  </span>
                </FormattedMessage>
              </p>
              <p
                className="mdc-typography--subtitle2"
              >
                Albany
              </p>
            </div>
          </div>
        </RepoHeader>
      </InjectIntl(RepoHeader)>
      <div
        className="repohome-body"
      >
        <InjectIntl(SearchBar)
          initialValue=""
          onSearch={[Function]}
          repoId="albany"
        >
          <SearchBar
            initialValue=""
            intl={
              Object {
                "defaultFormats": Object {},
                "defaultLocale": "en",
                "formatDate": [Function],
                "formatHTMLMessage": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelative": [Function],
                "formatTime": [Function],
                "formats": Object {},
                "formatters": Object {
                  "getDateTimeFormat": [Function],
                  "getMessageFormat": [Function],
                  "getNumberFormat": [Function],
                  "getPluralFormat": [Function],
                  "getRelativeFormat": [Function],
                },
                "locale": "en",
                "messages": Object {},
                "now": [Function],
                "onError": [Function],
                "textComponent": "span",
                "timeZone": null,
              }
            }
            onSearch={[Function]}
            repoId="albany"
          >
            <TextField
              className="searchbar"
              dense={false}
              floatingLabelClassName=""
              fullWidth={false}
              label="Search for a person"
              lineRippleClassName=""
              notchedOutlineClassName=""
              outlined={true}
              textarea={false}
            >
              <div
                className="mdc-text-field searchbar mdc-text-field--outlined"
                key="text-field-container"
                onClick={[Function]}
                onKeyDown={[Function]}
              >
                <Input
                  className=""
                  disabled={false}
                  foundation={
                    MDCTextFieldFoundation {
                      "adapter_": Object {
                        "activateLineRipple": [Function],
                        "addClass": [Function],
                        "closeOutline": [Function],
                        "deactivateLineRipple": [Function],
                        "deregisterInputInteractionHandler": [Function],
                        "deregisterTextFieldInteractionHandler": [Function],
                        "deregisterValidationAttributeChangeHandler": [Function],
                        "floatLabel": [Function],
                        "getLabelWidth": [Function],
                        "getNativeInput": [Function],
                        "hasClass": [Function],
                        "hasLabel": [Function],
                        "hasOutline": [Function],
                        "isFocused": [Function],
                        "notchOutline": [Function],
                        "registerInputInteractionHandler": [Function],
                        "registerTextFieldInteractionHandler": [Function],
                        "registerValidationAttributeChangeHandler": [Function],
                        "removeClass": [Function],
                        "setLineRippleTransformOrigin": [Function],
                        "shakeLabel": [Function],
                      },
                      "characterCounter_": undefined,
                      "helperText_": undefined,
                      "inputBlurHandler_": [Function],
                      "inputFocusHandler_": [Function],
                      "inputInputHandler_": [Function],
                      "isFocused_": false,
                      "isValid_": true,
                      "leadingIcon_": undefined,
                      "receivedUserInput_": false,
                      "setPointerXOffset_": [Function],
                      "textFieldInteractionHandler_": [Function],
                      "trailingIcon_": undefined,
                      "useNativeValidation_": true,
                      "validationAttributeChangeHandler_": [Function],
                      "validationObserver_": undefined,
                    }
                  }
                  handleFocusChange={[Function]}
                  handleValueChange={[Function]}
                  id=""
                  inputType="input"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  setDisabled={[Function]}
                  setInputId={[Function]}
                  syncInput={[Function]}
                  value=""
                >
                  <input
                    className="mdc-text-field__input"
                    disabled={false}
                    id=""
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    value=""
                  />
                </Input>
                <NotchedOutline
                  className=""
                  notch={false}
                  notchWidth={0}
                >
                  <div
                    className="mdc-notched-outline mdc-notched-outline--upgraded"
                  >
                    <div
                      className="mdc-notched-outline__leading"
                    />
                    <div
                      className="mdc-notched-outline__notch"
                      style={
                        Object {
                          "width": undefined,
                        }
                      }
                    >
                      <FloatingLabel
                        className=""
                        float={false}
                        handleWidthChange={[Function]}
                        htmlFor=""
                      >
                        <label
                          className="mdc-floating-label"
                          htmlFor=""
                          onAnimationEnd={[Function]}
                        >
                          Search for a person
                        </label>
                      </FloatingLabel>
                    </div>
                    <div
                      className="mdc-notched-outline__trailing"
                    />
                  </div>
                </NotchedOutline>
              </div>
            </TextField>
          </SearchBar>
        </InjectIntl(SearchBar)>
        <p
          className="mdc-typography--body1 repohome-recordcount"
        >
          <FormattedMessage
            defaultMessage="Currently tracking {recordCount} records"
            id="RepoHome.repoRecordCount"
            values={
              Object {
                "recordCount": 100,
              }
            }
          >
            <span>
              Currently tracking 100 records
            </span>
          </FormattedMessage>
        </p>
        <EndBarHeader>
          <div
            className="endbars-headerline-wrapper"
            dir="ltr"
          >
            <span
              className="mdc-typography--overline endbars-headerline"
            >
              <FormattedMessage
                defaultMessage="Or"
                id="RepoHome.or"
                values={Object {}}
              >
                <span>
                  Or
                </span>
              </FormattedMessage>
            </span>
          </div>
        </EndBarHeader>
        <RippledComponent
          className="pf-button-secondary"
          disabled={false}
          initRipple={[Function]}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          raised={true}
          style={Object {}}
          unbounded={false}
        >
          <Component
            className="pf-button-secondary"
            disabled={false}
            initRipple={[Function]}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            raised={true}
            style={Object {}}
          >
            <button
              className="mdc-button pf-button-secondary mdc-button--raised"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={Object {}}
            >
              <span
                className="mdc-button__label"
              >
                Provide information about someone
              </span>
            </button>
          </Component>
        </RippledComponent>
      </div>
      <InjectIntl(Footer)>
        <Footer
          intl={
            Object {
              "defaultFormats": Object {},
              "defaultLocale": "en",
              "formatDate": [Function],
              "formatHTMLMessage": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelative": [Function],
              "formatTime": [Function],
              "formats": Object {},
              "formatters": Object {
                "getDateTimeFormat": [Function],
                "getMessageFormat": [Function],
                "getNumberFormat": [Function],
                "getPluralFormat": [Function],
                "getRelativeFormat": [Function],
              },
              "locale": "en",
              "messages": Object {},
              "now": [Function],
              "onError": [Function],
              "textComponent": "span",
              "timeZone": null,
            }
          }
        >
          <div
            className="footer"
          >
            <p
              className="mdc-typography--body1"
            >
              <FormattedMessage
                defaultMessage="PLEASE NOTE: All data entered is available to the public and usable by anyone. Google does not review or verify the accuracy of this data. Google may share the data with public and private organizations participating in disaster response efforts."
                id="Footer.disclaimerText"
                values={Object {}}
              >
                <span>
                  PLEASE NOTE: All data entered is available to the public and usable by anyone. Google does not review or verify the accuracy of this data. Google may share the data with public and private organizations participating in disaster response efforts.
                </span>
              </FormattedMessage>
            </p>
          </div>
        </Footer>
      </InjectIntl(Footer)>
    </div>
  </RepoHome>
</InjectIntl(RepoHome)>
`;
